compiler/src/main/scala/org/scalajs/nscplugin/ExplicitInnerJS.scala

```diff
@@ -168,21 +175,22 @@ abstract class ExplicitInnerJS[G <: Global with Singleton](val global: G)
 
     /** The main transformation method. */
     override def transform(tree: Tree): Tree = {
-      val sym = tree.symbol
       tree match {
         // Add the ValDefs for inner JS class values
-        case Template(parents, self, decls) =>
+        case Template(parents, self, decls) if isApplicableOwner(currentOwner) =>
           val newDecls = mutable.ListBuffer.empty[Tree]
           atOwner(tree, currentOwner) {
             for (decl <- decls) {
-              if ((decl.symbol ne null) && isInnerJSClass(decl.symbol)) {
-                val clazz = decl.symbol
-                val jsclassAccessor = jsclassAccessorFor(clazz)
+              val declSym = decl.symbol
+              if (declSym eq null) {
+                // not a member def, do nothing
+              } else if (isJSClass(declSym)) {
```

Comment:
I reverted it in this commit, as indeed here it's not clear. But I reintroduced in the second commit anyway, because there I also have the check the Module case, so if I put `declSym ne null` in the condition, I have to reiterate it in the module condition as well.

Reply:
Right :) that makes sense.

I still find the empty `if` branch a bit strange. Maybe moving it in the `for`-loop...

```scala
for {
  decl <-
  declSym = decl.symbol
  if declSym ne null  // if declSym eq null, it's not a member def
} yield ...
```

Honestly, I'm not sure. I think this is OK as well.

Reply:
I can't move it in the for loop. I still need the last statement of the loop (`newDecls += decl`).